pkg/middleware/recovery/recovery_test.go

```diff
@@ -0,0 +1,161 @@
+package recovery
+
+import (
+	"context"
+	"net"
+	"net/http"
+	"net/http/httptest"
+	"testing"
+
+	grpc_recovery "github.com/grpc-ecosystem/go-grpc-middleware/recovery"
+	"github.com/stretchr/testify/require"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/credentials/insecure"
+	"google.golang.org/grpc/status"
+	"google.golang.org/grpc/test/bufconn"
+
+	openfgav1 "github.com/openfga/api/proto/openfga/v1"
+
+	"github.com/openfga/openfga/pkg/logger"
+)
+
+func TestPanic(t *testing.T) {
+	panicHandlerFunc := http.HandlerFunc(func(w http.ResponseWriter, req *http.Request) {
+		panic("Unexpected error!")
+	})
+
+	handler := HTTPPanicRecoveryHandler(panicHandlerFunc, logger.MustNewLogger("text", "info", "unix"))
+
+	req, err := http.NewRequest(http.MethodGet, "/", nil)
+	require.NoError(t, err)
+
+	resp := httptest.NewRecorder()
+	require.NotPanics(t, func() {
+		handler.ServeHTTP(resp, req)
+	})
+
+	require.Equal(t, http.StatusInternalServerError, resp.Code)
+}
+
+func TestUnaryPanicInterceptor(t *testing.T) {
+	listner := bufconn.Listen(1024 * 1024)
+	t.Cleanup(func() {
+		listner.Close()
+	})
+
+	serverOpts := []grpc.ServerOption{
+		grpc.ChainUnaryInterceptor(
+			[]grpc.UnaryServerInterceptor{
+				grpc_recovery.UnaryServerInterceptor(
+					grpc_recovery.WithRecoveryHandlerContext(
+						PanicRecoveryHandler(logger.MustNewLogger("text", "info", "unix")),
+					),
+				),
+			}...,
+		),
+	}
+
+	srv := grpc.NewServer(serverOpts...)
+	t.Cleanup(srv.Stop)
+
+	openfgav1.RegisterOpenFGAServiceServer(srv, &unimplementedOpenFGAServiceServer{})
+
+	go func() {
+		err := srv.Serve(listner)
+		if err != nil {
```

Comment:
```suggestion
		require.NoError(t, err)
```
instead of `if err != nil { t.Fail }` please.

Use the `require` package where possible.

Reply:
go-require: require must only be used in the goroutine running the test function (testifylint)